tests/xtro-sharpie/SelectorCheck.cs

```diff
@@ -51,24 +33,53 @@ public override void VisitManagedMethod (MethodDefinition method)
 			foreach (var ca in method.CustomAttributes) {
 				switch (ca.Constructor.DeclaringType.Name) {
 				case "ExportAttribute":
-					string selector = ca.ConstructorArguments [0].Value as string;
-					if (!known_selectors.Contains (selector))
-						known_selectors.Add (selector);
+					//string selector = ca.ConstructorArguments [0].Value as string;
+					//if (!known_selectors.Contains (selector))
+					//known_selectors.Add (selector);
+
+					var methodDefinition = method.GetName ();
+					if (!string.IsNullOrEmpty (methodDefinition)) {
+						var argumentSemantic = Helpers.ArgumentSemantic.None;
+						if (ca.ConstructorArguments.Count > 1) {
+							argumentSemantic = (Helpers.ArgumentSemantic)ca.ConstructorArguments [1].Value;
+						}
+
+						qualified_export_arguments.Add ((methodDefinition, argumentSemantic));
+					}
 
-					qualified_selectors.Add (method.GetName ());
-//
-//					TypeDefinition type = method.DeclaringType;
-//					HashSet<string> list;
-//					if (!type_exports.TryGetValue (type, out list)) {
-//						list = new HashSet<string> ();
-//						type_exports.Add (type, list);
-//					}
-//					list.Add (selector);
 					break;
 				}
 			}
 		}
 
+		public override void VisitObjCPropertyDecl (ObjCPropertyDecl decl)
+		{
+			// protocol members are checked in ObjCProtocolCheck
+			if (decl.DeclContext is ObjCProtocolDecl)
+				return;
+
+			// check availability macros to see if the API is available on the OS and not deprecated
+			if (!decl.IsAvailable ())
+				return;
+
+			var framework = Helpers.GetFramework (decl);
+			if (framework == null)
+				return;
+
+			var nativeArgumentSemantic = decl.Attributes.ToArgumentSemantic ();
+
+			// Ignore Copy, UnsafeUnretained and Weak for now
+			if (nativeArgumentSemantic == Helpers.ArgumentSemantic.Copy || nativeArgumentSemantic == Helpers.ArgumentSemantic.UnsafeUnretained || nativeArgumentSemantic == Helpers.ArgumentSemantic.Weak)
+				return;
+
+			var nativeMethodDefinition = decl.QualifiedName;
+
+			var exportArgs = qualified_export_arguments.FirstOrDefault (sel => sel.MethodDefinition.Contains (nativeMethodDefinition));
+
+			if (!string.IsNullOrEmpty (exportArgs.MethodDefinition) && exportArgs.ArgumentSemantic != nativeArgumentSemantic)
+				Log.On (framework).Add ($"!incorrect-argument-semantic! {nativeMethodDefinition} has ArgumentSemantic.{nativeArgumentSemantic.ToUsableString ()} instead of ArgumentSemantic.{exportArgs.ArgumentSemantic.ToUsableString ()}");
```

Comment:
```
!incorrect-argument-semantic! SKUniform::textureValue has ArgumentSemantic.Strong|Retain instead of ArgumentSemantic.None
```
it's a bit confusing when reading (without our bindings sources or xtro sources). 

Either (best) you use the `ArgumentSemantic` for the managed version and the ObjC syntax (for clang/headers); or you mention which one is managed and native.

Reply:
@spouliot `!incorrect-argument-semantic! 'SKUniform::textureValue' (native) has (strong|retain) instead of 'ArgumentSemantic.None'`?

Reply:
@VincentDondain yes, that would remove the ambiguity :)